restli-int-test/src/test/java/com/linkedin/restli/examples/TestParseqBasedFluentClientApi.java

```diff
@@ -283,6 +291,362 @@ public void testBatchCreateReturnEntityDisabled() throws InterruptedException, E
     Assert.assertEquals(ids.size(), 2);
   }
 
+  @Test
+  public void testCreateAndThenBatchDelete() throws InterruptedException, ExecutionException
+  {
+    Greetings greetings = new Greetings(_parSeqRestliClient, _parSeqUnitTestHelper.getEngine());
+    CountDownLatch latch = new CountDownLatch(1);
+    // Create entities first so we don't delete those used by other tests.
+    CompletionStage<List<CreateIdStatus<Long>>>
+        createResult = greetings.batchCreate(Arrays.asList(getGreeting(), getGreeting()));
+
+    CompletionStage<Map<Long, UpdateStatus>>
+        result = createResult.thenCompose(ids -> greetings.batchDelete(
+            Sets.newHashSet(ids.stream().map(CreateIdStatus::getKey).collect(Collectors.toList()))));
+    result.whenComplete((r, t)-> latch.countDown());
+    Assert.assertTrue(latch.await(5000, TimeUnit.MILLISECONDS),
+        "Request timed out");
+    CompletableFuture<Map<Long, UpdateStatus>> future = result.toCompletableFuture();
+    Assert.assertTrue(future.isDone());
+    Map<Long, UpdateStatus> ids = future.get();
+    Assert.assertEquals(ids.size(), 2);
+    for (UpdateStatus status : ids.values())
+    {
+      Assert.assertEquals(status.getStatus().intValue(), 204);
+    }
+  }
+
+  @Test
+  public void testCreateAndThenBatchDeleteWithFailures() throws InterruptedException, ExecutionException
+  {
+    Greetings greetings = new Greetings(_parSeqRestliClient, _parSeqUnitTestHelper.getEngine());
+    CountDownLatch latch = new CountDownLatch(1);
+    // Create entities first so we don't delete those used by other tests.
+    CompletionStage<List<CreateIdStatus<Long>>>
+        createResult = greetings.batchCreate(Arrays.asList(getGreeting(), getGreeting()));
+
+    CompletionStage<Map<Long, UpdateStatus>>
+        result = createResult.thenCompose(ids ->
+    {
+      Set<Long> deleteIds = Sets.newHashSet(
+          ids.stream().map(CreateIdStatus::getKey).collect(Collectors.toList()));
+      deleteIds.add(-1L);
+      return greetings.batchDelete(deleteIds);
+    });
+    result.whenComplete((r, t)-> latch.countDown());
```

Comment:
Just curious on using the countDownLatch, can we just await on the result directly?

Reply:
Good point, fixed it. I was using count down latch as I was directly checking CompletionStage before (it doesn't have get with timeout).